### PR TITLE
Message Window

### PR DESCRIPTION
**Pull request by Svelbeard**
Message Window.

Also added stub for News Window.

### CODE REVIEW

**Comment by Svelbeard** [comment]
I've just noticed that when the window resizes from the message tab to the settings tab it leaves traces of the original window size.

File: src/openloco/windows/MessageWindow.cpp [hunk 1]
```diff
@@ -1,26 +1,683 @@
...
     // 0x0042A3FF
     void open()
     {
-        call(0x0042A3FF);
+        //call(0x0042A3FF);
+        auto window = WindowManager::bringToFront(WindowType::messages);
+
+        if (window != nullptr)
+        {
+            if (input::is_tool_active(_toolWindowType, _toolWindowNumber))
```
Comment (duncanspumpkin) [inline src/openloco/windows/MessageWindow.cpp:305]: This doesn't look correct. I think you are meant to pass the window->class and window->number

File: src/openloco/windows/MessageWindow.cpp [hunk 2]
```diff
@@ -1,26 +1,683 @@
...
+            window->flags |= window_flags::resizable;
+
+            window->owner = _playerCompany;
+            auto skin = objectmgr::get<interface_skin_object>();
+            window->colours[1] = skin->colour_0A;
+
+            window->width = messages::minWindowSize.width;
+            window->height = messages::minWindowSize.height;
+
+            window->invalidate();
```
Comment (duncanspumpkin) [inline src/openloco/windows/MessageWindow.cpp:347]: From this point onwards it should merge with the window != nullptr code

File: src/openloco/windows/MessageWindow.cpp [hunk 3]
```diff
@@ -1,26 +1,683 @@
...
+
+            window->widgets = messages::widgets;
+            window->holdable_widgets = 0;
```
Comment (duncanspumpkin) [inline src/openloco/windows/MessageWindow.cpp:350]: enabled_widgets set is missing

File: src/openloco/windows/MessageWindow.cpp [hunk 4]
```diff
@@ -1,26 +1,683 @@
...
+            self->widgets[common::widx::panel].right = self->width - 1;
+            self->widgets[common::widx::panel].bottom = self->height - 1;
+
+            self->widgets[common::widx::caption].right = self->width - 2;
+
+            self->widgets[common::widx::close_button].left = self->width - 15;
+            self->widgets[common::widx::close_button].right = self->width - 3;
+        }
+
+        static void switchTab(window* self, widget_index widgetIndex)
```
Comment (duncanspumpkin) [inline src/openloco/windows/MessageWindow.cpp:609]: add a loco offset `0x42A716`

File: src/openloco/windows/MessageWindow.cpp [hunk 5]
```diff
@@ -1,26 +1,683 @@
...
+            int32_t pan = self->width / 2 + self->x;
+            audio::play_sound(audio::sound_id::click_down, pan);
+        }
+
+        // 0x0042A87C
+        static void scroll_mouse_over(ui::window* self, int16_t x, int16_t y, uint8_t scrollIndex)
+        {
+            self->flags &= ~(window_flags::not_scroll_view);
+
+            auto messageIndex = y / 39;
```
Comment (duncanspumpkin) [inline src/openloco/windows/MessageWindow.cpp:179]: define 39 as the message height its used in a bunch of these functions

File: src/openloco/windows/MessageWindow.cpp [hunk 6]
```diff
@@ -1,26 +1,683 @@
...
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+        }
+
+        // 0x0042A5D7
+        static void draw_scroll(ui::window* self, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+        {
+            auto colour = colour::get_shade(self->colours[1], 4);
+            auto fill = colour << 24 | colour << 16 | colour << 8 | colour;
```
Comment (duncanspumpkin) [inline src/openloco/windows/MessageWindow.cpp:218]: use `clear_single` then you don't need to do this.

**Comment by duncanspumpkin** [comment]
Could you rebase onto master and this is ready to go.

**Pull request merged**